### DISCUSSION TITLE
Is the use of extension mandatory?

### DISCUSSION THREAD

**Discussion by Fthayy** [hoppscotch/hoppscotch#3040] [Q&A] [answered]
I'm trying to run hoppscotch locally and I need to use extension for that. Is it possible to do that without extensions. If I can, how can I do that? https://github.com/hoppscotch/hoppscotch/discussions/2051#discussioncomment-2721117 I found the same question here but I coulnd't find the answer.

**Comment by AndrewBastin** [comment, marked as answer]
@Fthayy You do not need the extension if you do not need to hit endpoints which fail the CORS test that browsers run. Hoppscotch is a web app like every other app so when requests go out to 3rd party origins, browsers run a CORS test (see https://developer.mozilla.org/en-US/docs/Web/Security/Same-origin_policy) to see if the operation is allowed by the Same Origin Policy.

The extension bypasses this by running requests in the extension (to which the Same Origin Policy does not apply) and return back the data from that.

Additionally, you can use the Proxy Interceptor to run the query for any publicly available endpoints (no local intranets or localhost) without the extension.

Hope this answers your question
